src/core/common/settings.cpp

```diff
@@ -196,8 +196,8 @@ const uint16_t Settings::kCriticalKeys[] = {
 
 void Settings::Init(void)
 {
-    Get<SettingsDriver>().Init();
     Get<SettingsDriver>().SetCriticalKeys(kCriticalKeys, GetArrayLength(kCriticalKeys));
+    Get<SettingsDriver>().Init();
```

Comment:
Should we pass critical keys as the arguments of Init?

Reply:
+1 on @simonlingoogle suggestion.

Requiring `Init()` to be called first before calling any other method on `SettingDriver` seems reasonable to me.

Reply:
+1

Reply:
SGTM. Updated. I also added them as parameters of `otPlatSettingsInit` so that `otPlatSettingsSetCriticalKeys` is no longer needed.